functions/nodejs-apps.bash

```diff
@@ -342,3 +342,60 @@ zigbee2mqtt_setup() {
     whiptail --title "Operation successful" --msgbox "$installSuccessText" 15 80
   fi
 }
+
+## Function for installing a npm package for the JS Scripting Automation Add-On
+##
+##    jsscripting_npm_install(String packageName, String mode)
+##    Available values for mode: "update". Defaults to "install".
+##
+jsscripting_npm_install() {
+  if [ "${1}" == "" ]; then echo "FAILED. Provide packageName."; return 1; fi
+
+  if ! node_is_installed || is_armv6l; then
+    echo -n "$(timestamp) [openHABian] Installing prerequsites for ${1} (NodeJS)... "
+    if cond_redirect nodejs_setup; then echo "OK"; else echo "FAILED"; return 1; fi
+  fi
+
+  if [ "${2}" == "update" ];
+  then
+    echo -n "$(timestamp) [openHABian] Updating ${1} for JS Scripting... "
+    if cond_redirect sudo -u "${username:-openhabian}" npm update --prefix "/etc/openhab/automation/js" "${1}"; then echo "OK"; else echo "FAILED (npm update)"; return 1; fi
```

Comment:
Do we have to worry about file permissions here? At some point openHAB will change the ownership of everything in `/etc/openhab/` to `openhab:openhab`. It might cause problems to update/install as user `openhabian` or it might cause problems for openHAB after the update is complete.

Reply:
That‘s a good point, thanks!
I am nut sure if this can cause trouble (depends on the chmod number), but even if not, it is the better way to install as openhab:openhab.
Why should we install as `openhabian` when it anyway gets modified to `openhab:openhab` later?